Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java

```diff
@@ -0,0 +1,301 @@
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azuretools.core.mvp.model.mysql;
+
+import com.google.common.base.Preconditions;
+import com.microsoft.azure.management.mysql.v2020_01_01.Server;
+import com.microsoft.azure.management.mysql.v2020_01_01.ServerPropertiesForDefaultCreate;
+import com.microsoft.azure.management.mysql.v2020_01_01.ServerState;
+import com.microsoft.azure.management.mysql.v2020_01_01.ServerUpdateParameters;
+import com.microsoft.azure.management.mysql.v2020_01_01.ServerVersion;
+import com.microsoft.azure.management.mysql.v2020_01_01.SslEnforcementEnum;
+import com.microsoft.azure.management.mysql.v2020_01_01.implementation.DatabaseInner;
+import com.microsoft.azure.management.mysql.v2020_01_01.implementation.FirewallRuleInner;
+import com.microsoft.azure.management.mysql.v2020_01_01.implementation.MySQLManager;
+import com.microsoft.azure.management.resources.Subscription;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azuretools.authmanage.AuthMethodManager;
+import com.microsoft.azuretools.core.mvp.model.AzureMvpModel;
+import lombok.Lombok;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MySQLMvpModel {
+
+    private static final String NAME_ALLOW_ACCESS_TO_AZURE_SERVICES = "AllowAllWindowsAzureIps";
+    private static final String IP_ALLOW_ACCESS_TO_AZURE_SERVICES = "0.0.0.0";
+    private static final String NAME_PREFIX_ALLOW_ACCESS_TO_LOCAL = "ClientIPAddress_";
+    private static final List<String> MYSQL_SUPPORTED_REGIONS = Arrays.asList(
+            "australiacentral", "australiacentral2", "australiaeast", "australiasoutheast", "brazilsouth", "canadacentral", "canadaeast", "centralindia",
```

Comment:
do we have api to get the supported regions?

Reply:
have no region api for mysql.